### PR TITLE
Updated the installation instructions for macOS

### PR DESCRIPTION
**Pull request by asunapg**
Added: Instructions on changing directories after the repo clone (cd bitcoin)
(Rationale: The initial documentation assumes that the user knows to change the current directory to the cloned repository before running further commands. However, for new users or those unfamiliar with the command line, this step might not be obvious.)

Added: Instructions on verifying the build was successful and the GUI was compiled.
(Rationale: After compiling the Bitcoin Core software, it's important to verify that the build was successful and that the GUI, if compiled, is functional. This step helps users confirm that the installation process is complete and that the software is ready to use.)

### CODE REVIEW

File: doc/build-osx.md [hunk 1]
```diff
@@ -60,6 +60,11 @@ All build scripts and commands will run from this directory.
 ``` bash
 git clone https://github.com/bitcoin/bitcoin.git
 ```
+From here, change your working directory to "bitcoin" by running
```
Comment (Riahiamirreza) [inline doc/build-osx.md:63]: nit, using colon at the end of the line may looks better

> From here, change your working directory to "bitcoin" by running:

**Comment by RandyMcMillan** [comment]
ACK

C8B1233D-4AA6-4E7B-BCA2-6DB0C9F8F9B1
A75F866E-A552-41CD-8BF2-54B99D43E0AD

**Comment by RandyMcMillan** [comment]
Note: it is good idea to check that markdown looks good in desktop editors (macdown for example) - github flavored markdown has diverged from other flavors and sometimes there are minor issues.

**Comment by fanquake** [comment]
Thanks, however we are going to leave this as-is for now. Generic changes to build docs are better suited if PR'd against all platforms, not sure we need to make the other additions.

**Pull request closed without merging**